src/main.js

```diff
@@ -244,3 +244,81 @@ let destinations = [
   ]
 
 
+  //visited and bucktelist lists
+  const visited = destinations.filter(item => item.visited === true);
+  const bucketlist = destinations.filter(item => item.visited === false);
+ 
+
+  //get list of urls
+  const getURL = (destinations) => {
+    let url= destinations.map(obj => obj.photo.valueOf());
+    return url;
+  }
+
+  //create elements for each city
+  const elements= destinations.forEach(destination => {
+
+    //picture element
+    const createImg = (url) => {
+      let figure = document.createElement('figure');
+    //img element
+      let img = document.createElement('img');
+      img.style.objectFit ='cover';
+      document.body.appendChild(figure);
+      figure.appendChild(img);
+    //img caption
+      let figcaption= document.createElement('figcaption');
+      figure.appendChild(figcaption);
+      figcaption.innerText= `${destinations.city}, ${destinations.country}`
+
+
+      //get urls
+      for(let i=0; i<destinations.length; i++) {
+        img.src = url[i];
+      };
+      
+      //add classes 
+      if(destination.visited === true) {
+        img.classList.add('visited');
+      }
+      else {
+        img.classList.add('bucketlist');
+      }
+
+  };
+
+
+
+  return createImg(getURL(destinations));
+
+});
+
+  //create buttons
+  const buttonVisited = document.createElement('button');
+  buttonVisited.innerText='Visited';
+  buttonVisited.type='button';
+  document.body.appendChild(buttonVisited);
```

Comment:
> Also I'm still getting ReferenceError: document is not defined when running code in editor.

Where exactly do you get that error? Can you show a screenshot and the exact steps? You are running the project with `npm run dev`, right?

Reply:
It's just in the editor when I try to use an extension to run js (with npm running)
<img width="1040" alt="Screenshot 2020-09-05 at 16 55 11" src="https://user-images.githubusercontent.com/62510606/92307735-c953c800-ef98-11ea-9151-1d42a29a0d17.png">

Reply:
"I try to use an extension to run js (with npm running)". What extension? What do you use it for?

Reply:
This one, I just used it to run js in vscode instead of browser console 
<img width="469" alt="Screenshot 2020-09-05 at 17 42 36" src="https://user-images.githubusercontent.com/62510606/92308627-57cb4800-ef9f-11ea-8ddf-6dd931fcef53.png">

Reply:
Oh, but this is to run server-side code, and there document does not exist (it's something that only exists in browsers).
Anyway, here, you should run the code in developer tools and learn how to use debugger and breakpoints there. It will help u a lot debugging the code ;) [Here's a chrome tutorial](https://www.youtube.com/watch?v=H0XScE08hy8&ab_channel=GoogleChromeDevelopers) (it also works for firefox!)